File: codility/Iterations_BinaryGap.py
def solution(N):
    binary = bin(N)[2:]
    flag = 0
    cnt = 0
    max_cnt = 0
    for idx, val in enumerate(binary):
        if val == '1':
            if flag and cnt != 0:
                max_cnt = max(cnt, max_cnt)
                cnt = 0
            else:
                flag = 1
        else: # val == '0'
            if flag:
                cnt += 1
    return max_cnt

File: codility/Iterations_BinaryGap.java
import java.util.*;

class Solution {
    public int solution(int N) {
        String binary = Integer.toBinaryString(N);
        int answer = 0;
        int cnt = 0;
        for (int i = 1; i < binary.length(); i++) {
            if (binary.charAt(i) == '0') {
                cnt += 1;
            }
            else {
                if (cnt > answer) {
                    answer = cnt;
                }
                cnt = 0;
            }
        }
        return answer;
    }
}

// https://app.codility.com/demo/results/trainingC63HF5-PQ3/
